### ISSUE TITLE
Feature Request: Allow Yarn to log unexpected error mesages to stderr

### ISSUE THREAD

**Issue by Taytay** [yarnpkg/yarn#1456]
**Do you want to request a _feature_ or report a _bug_?**
Feature request

**What is the current behavior?**
When Yarn gets an unexpected error, it currently tells me:
`error An unexpected error occured, please open a bug report with the information provided in "<snip>.../yarn-error.log".`

I am getting these errors on our build machine, which means that our build logs don't have the text of the error, so now I need to log into the machine to dump that file. So far, I can always identify the relevant bit of the error from the "Trace" message at the very bottom of the file, like so:

```
Trace:
  Error: EACCES: permission denied, open '/home/vagrant/.yarn/.roadrunner.json'
      at Error (native)
      at Object.fs.openSync (fs.js:549:18)
      at Object.fs.writeFileSync (fs.js:1156:15)
      at /usr/lib/node_modules/yarn/node_modules/roadrunner/index.js:25:6
      at /usr/lib/node_modules/yarn/node_modules/roadrunner/index.js:12:12
      at emitOne (events.js:82:20)
      at process.emit (events.js:169:7)
      at processEmit [as emit] (/usr/lib/node_modules/yarn/node_modules/loud-rejection/node_modules/signal-exit/index.js:140:35)
      at process.exit (node.js:750:17)
      at exit (/usr/lib/node_modules/yarn/lib-legacy/cli/index.js:380:13)
```

It would be very helpful in our case to log this information, (or maybe even just the message without the stack?) to stderr so that I wouldn't have to dig somewhere else to get error details.

So perhaps the message could change to: (Spelling of occurred corrected here as well:)

```
error An unexpected error occurred please open a bug report with the information provided in "<snip>.../yarn-error.log".
Trace:
 Error: EACCES: permission denied, open '/home/vagrant/.yarn/.roadrunner.json'
      at Error (native)
      at Object.fs.openSync (fs.js:549:18)
      at Object.fs.writeFileSync (fs.js:1156:15)
      at /usr/lib/node_modules/yarn/node_modules/roadrunner/index.js:25:6
      at /usr/lib/node_modules/yarn/node_modules/roadrunner/index.js:12:12
      at emitOne (events.js:82:20)
      at process.emit (events.js:169:7)
      at processEmit [as emit] (/usr/lib/node_modules/yarn/node_modules/loud-rejection/node_modules/signal-exit/index.js:140:35)
      at process.exit (node.js:750:17)
      at exit (/usr/lib/node_modules/yarn/lib-legacy/cli/index.js:380:13)
```

**Comment by Daniel15** [comment]
What if we output just the message but not the stack trace? Seems like that'd be sufficient. We can continue to hide the stack trace in the log file.

**Comment by Taytay** [comment]
That sounds great to me @Daniel15. It's often sufficient and succinct.

**Comment by jvhoven** [comment]
Hi, I would like to try and implement this. Would something like the following suffice?

```
error An unexpected error occurred: "Error: EACCES: permission denied, open '/home/jeffrey/Documents/Projects/yarn-testing/package.json'"
info If you think this is a bug, please open a bug report with the information provided in "/home/jeffrey/Documents/Projects/yarn-testing/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
```

**Comment by kittens** [comment]
Closed via #1502.

**Comment by Taytay** [comment]
Thanks @jvhoven and @kittens !
